### PR TITLE
Replace beta_fr_ema with kappa for eligibility trace filtering

### PR DESCRIPTION
**Pull request by JesusEV**
Previously, models utilized a parameter named `beta_fr_ema` to control the low-pass filtering of the firing rate regularization term. This update replaces `beta_fr_ema` with `kappa` which standardize the treatment of both the learning signal and firing rate regularization contributions, thereby simplifying notation and saving computations.

### CODE REVIEW

File: models/eprop_iaf.cpp [hunk 1]
```diff
@@ -190,10 +190,9 @@ eprop_iaf::Parameters_::set( const DictionaryDatum& d, Node* node )
     throw BadProperty( "Spike threshold voltage V_th ≥ minimal voltage V_min required." );
   }
 
-  if ( beta_fr_ema_ < 0 or 1 <= beta_fr_ema_ )
+  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
   {
-    throw BadProperty(
-      "Smoothing factor of firing rate exponential moving average beta_fr_ema_ from interval [0,1) required." );
+    throw BadProperty( "kappa must be in the range (0, 1)" );
```
Comment (akorgor) [inline models/eprop_iaf.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```

File: models/eprop_iaf_adapt.cpp [hunk 2]
```diff
@@ -210,10 +210,9 @@ eprop_iaf_adapt::Parameters_::set( const DictionaryDatum& d, Node* node )
     throw BadProperty( "Spike threshold voltage V_th ≥ minimal voltage V_min required." );
   }
 
-  if ( beta_fr_ema_ < 0 or 1 <= beta_fr_ema_ )
+  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
   {
-    throw BadProperty(
-      "Smoothing factor of firing rate exponential moving average beta_fr_ema_ from interval [0,1) required." );
+    throw BadProperty( "kappa must be in the range (0, 1)" );
```
Comment (akorgor) [inline models/eprop_iaf_adapt.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```

File: models/eprop_iaf_psc_delta.cpp [hunk 3]
```diff
@@ -207,10 +207,9 @@ nest::eprop_iaf_psc_delta::Parameters_::set( const DictionaryDatum& d, Node* nod
     throw BadProperty( "Firing rate regularization target rate f_target ≥ 0 required." );
   }
 
-  if ( beta_fr_ema_ < 0 or 1 <= beta_fr_ema_ )
+  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
   {
-    throw BadProperty(
-      "Smoothing factor of firing rate exponential moving average beta_fr_ema_ from interval [0,1) required." );
+    throw BadProperty( "kappa must be in the range (0, 1)" );
```
Comment (akorgor) [inline models/eprop_iaf_psc_delta.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```

**Review by akorgor — CHANGES_REQUESTED** [review]
Nice change! Please do the following before merging to make the checks pass:

- [x] run  `build_support/format_all_c_c++_files.sh`

File: models/eprop_iaf.cpp [hunk 4]
```diff
@@ -190,7 +190,7 @@ eprop_iaf::Parameters_::set( const DictionaryDatum& d, Node* node )
     throw BadProperty( "Spike threshold voltage V_th ≥ minimal voltage V_min required." );
   }
 
-  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
+  if ( kappa_ < 0.0 or kappa_ > 1.0 )
   {
     throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```
Comment (akorgor) [inline models/eprop_iaf.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range (0, 1) required." );
```

File: models/eprop_iaf_adapt.cpp [hunk 5]
```diff
@@ -210,7 +210,7 @@ eprop_iaf_adapt::Parameters_::set( const DictionaryDatum& d, Node* node )
     throw BadProperty( "Spike threshold voltage V_th ≥ minimal voltage V_min required." );
   }
 
-  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
+  if ( kappa_ < 0.0 or kappa_ > 1.0 )
   {
     throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```
Comment (akorgor) [inline models/eprop_iaf_adapt.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range (0, 1) required." );
```

File: models/eprop_iaf_psc_delta.cpp [hunk 6]
```diff
@@ -207,7 +207,7 @@ nest::eprop_iaf_psc_delta::Parameters_::set( const DictionaryDatum& d, Node* nod
     throw BadProperty( "Firing rate regularization target rate f_target ≥ 0 required." );
   }
 
-  if ( kappa_ <= 0.0 or kappa_ > 1.0 )
+  if ( kappa_ < 0.0 or kappa_ > 1.0 )
   {
     throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
```
Comment (akorgor) [inline models/eprop_iaf_psc_delta.cpp]:
```suggestion
    throw BadProperty( "Eligibility trace low-pass filter from range (0, 1) required." );
```

**Review by akorgor — CHANGES_REQUESTED** [review]
Given commit https://github.com/JesusEV/nest-simulator/pull/1/commits/e709c2d5ceb5e787d5126d5528ed3dd33011c7b2 the ranges in the `BadProperty` error have to be adjusted.

File: models/eprop_iaf_psc_delta.cpp (hunk 6 above)
Comment (JesusEV) [reply models/eprop_iaf_psc_delta.cpp]: I think it should be corrected to:
```cpp
  if ( kappa_ < 0.0 or kappa_ >= 1.0 )
  {
    throw BadProperty( "Eligibility trace low-pass filter from range [0, 1) required." );
 ```

Comment (akorgor) [reply models/eprop_iaf_psc_delta.cpp:212]: Sorry, my attention slipped and I mixed up the notation for closed interval (correct is `[`) and open (correct is `(`). Maybe we could keep both open, so that we can test the two edge cases.

Comment (akorgor) [reply models/eprop_iaf_psc_delta.cpp]: The notation is `[` for closed, i.e., `<=` or `>=`, and `(` for open, i.e., `>` or `<`, so the condition you suggest would correspond to `(0, 1]`. However, maybe we should keep it open into both directions, i.e., `0 < kappa, kappa > 1` and `(0,1)` to be able to test the edge cases.

**Review by akorgor — CHANGES_REQUESTED** [review]
Great change with adding the firing rate regularization to the learning signal. I just have a follow-up comment on the kappa range.

File: models/eprop_iaf_psc_delta.cpp (hunk 6 above)
Comment (JesusEV) [reply models/eprop_iaf_psc_delta.cpp]: @akorgor That's not true... the condition `kappa_ < 0.0 or kappa_ >= 1.0`  implies a `[0,1)`. 

And for testing edge cases I guess you really meant `kappa < 0 , kappa > 1`, which implies `[0,1]`. Correct?

**Comment by JesusEV** [comment]
@akorgor last commit adds your suggestion to test edge cases

File: models/eprop_iaf_psc_delta.cpp (hunk 6 above)
Comment (akorgor) [reply models/eprop_iaf_psc_delta.cpp]: Please see the notation, e.g. here https://www.wikiwand.com/en/Interval_(mathematics)#Definitions_and_terminology

Comment (JesusEV) [reply models/eprop_iaf_psc_delta.cpp]: @akorgor I don't really understand why you are still discussing this.  I do not understand the confusion.

Comment (akorgor) [reply models/eprop_iaf_psc_delta.cpp]: Sorry, just figured out the confusion - I was misreading the condition. I now understand that the comparison signs indicate what we discard, so yes correct is `kappa_ < 0.0` and `kappa > 1` implying `[0,1]`.

**Review by akorgor — APPROVED** [review]
